common/include/camera/ICameraControl.hpp

```diff
@@ -0,0 +1,312 @@
+            bool& keyDown = m_keysDown[virtualKey];
+
+            using virtual_key_state_t = std::tuple<ui::E_KEY_CODE /*physical key representing virtual key*/, bool /*is pressed*/, float64_t /*delta action*/>;
+
+            auto updateVirtualState = [&]() -> virtual_key_state_t
+            {
+                virtual_key_state_t state = { ui::E_KEY_CODE::EKC_NONE, false, 0.f };
+
+                for (const auto& ev : events) // TODO: improve the search
+                {
+                    if (ev.keyCode == code)
+                    {
+                        if (ev.action == nbl::ui::SKeyboardEvent::ECA_PRESSED && !keyDown)
+                            keyDown = true;
+                        else if (ev.action == nbl::ui::SKeyboardEvent::ECA_RELEASED)
+                            keyDown = false;
+
+                        const auto dt = std::chrono::duration_cast<std::chrono::milliseconds>(m_nextPresentationTimeStamp - ev.timeStamp).count();
+                        assert(dt >= 0);
+
+                        state = std::make_tuple(code, keyDown, dt);
+                        break;
+                    }
+                }
+
+                return state;
+            };
+
+            const auto&& [physicalKey, isDown, dtAction] = updateVirtualState();
+
+            if (physicalKey != ui::E_KEY_CODE::EKC_NONE)
+                if (isDown)
+                    output.emplace_back(CVirtualEvent{ virtualKey, static_cast<float64_t>(dtAction) });
+        }
+
+        return output;
+    }
+
+    /*
+    // [OPTIONAL]: process mouse to generate virtual manipulation events
+    // note that:
+    // - all manipulations *may* be done with keyboard keys (if you have a touchpad or sth an ui:: event could be a code!)
+    // - it doesn't make the manipulation itself!
+    */
+    std::vector<CVirtualEvent> processMouse(std::span<const ui::SMouseEvent> events) const
+    {
+        double dPitch = {}, dYaw = {};
+
+        for (const auto& ev : events)
+            if (ev.type == nbl::ui::SMouseEvent::EET_MOVEMENT)
+            {
+                dYaw += ev.movementEvent.relativeMovementX;
+                dPitch += ev.movementEvent.relativeMovementY;
+            }
+
+        std::vector<CVirtualEvent> output;
+
+        if (dPitch)
+        {
+            auto& pitch = output.emplace_back();
+            pitch.type = dPitch > 0.f ? TiltUp : TiltDown;
+            pitch.value = std::abs(dPitch);
+        }
+
+        if (dYaw)
+        {
+            auto& yaw = output.emplace_back();
+            yaw.type = dYaw > 0.f ? PanRight : PanLeft;
+            yaw.value = std::abs(dYaw);
+        }
+
+        return output;
+    }
+
+    inline void setMoveSpeed(const float moveSpeed) { m_moveSpeed = moveSpeed; }
+    inline void setRotateSpeed(const float rotateSpeed) { m_rotateSpeed = rotateSpeed; }
+
+    inline const float getMoveSpeed() const { return m_moveSpeed; }
+    inline const float getRotateSpeed() const { return m_rotateSpeed; }
```

Comment:
not sure they belong in the "camera interface for all cameras"

Reply:
removed